### PR TITLE
feat(fortran): `SPC m f o` opens project config

### PR DESCRIPTION
**Pull request by fosskers**
This PR adds a new convenience function and binding to the `fortran` module.

-----
- [x] I searched the issue tracker and this hasn't been PRed before.
- [x] My changes are not on [the do-not-PR list](https://doomemacs.org/d/do-not-pr) for this project.
- [x] My commits conform to [the git conventions](https://doomemacs.org/d/git-conventions).
- [ ] I am blindly checking these off.

### CODE REVIEW

File: modules/lang/fortran/autoload.el [hunk 1]
```diff
@@ -115,3 +115,10 @@ or gfortran, depending on what feature flags are set."
   "Test the current project using fpm."
   (interactive)
   (compile "fpm test"))
+
+;;;###autoload
+(defun +fortran/fpm-open-project-toml ()
+  "Open the TOML file associated with the current project."
+  (interactive)
+  (when-let ((root (doom-project-root)))
+    (find-file (file-name-concat root "fpm.toml"))))
```
Comment (fosskers) [inline modules/lang/fortran/autoload.el:124]: Silently does nothing when not in an FPM-based project.

Comment (hlissner) [reply modules/lang/fortran/autoload.el:124]: To make it more versatile and improve its feedback, please change it to:

```elisp
(defun +fortran/fpm-open-project-toml (project-root)
  "Open fpm.toml at PROJECT-ROOT (defaults to the open project)."
  (interactive (list (doom-project-root)))
  (let ((file (file-name-concat project-root "fpm.toml")))
    (if (file-exists-p file)
        (find-file file)
      (user-error "No fpm.toml found at project root (%s)"
                  (abbreviate-file-name project-root)))))
```

**Review by hlissner — CHANGES_REQUESTED** [review]

File: modules/lang/fortran/autoload.el (hunk 1 above)
Comment (fosskers) [reply modules/lang/fortran/autoload.el:124]: That would be much better yeah. Thanks for reminding me about `user-error`!

File: modules/lang/fortran/autoload.el [hunk 2]
```diff
@@ -115,3 +115,13 @@ or gfortran, depending on what feature flags are set."
...
+
+;;;###autoload
+(defun +fortran/fpm-open-project-toml (project-root)
+  "Open fpm.toml at PROJECT-ROOT (defaults to the open project)."
+  (interactive (list (doom-project-root)))
+  (let ((file (file-name-concat project-root "fpm.toml")))
+    (cond ((file-exists-p file) (find-file file))
+          ((null project-root) (user-error "Not in a project"))
+          (t (user-error "No fpm.toml found at project root (%s)"
+                         (abbreviate-file-name project-root))))))
```
Comment (fosskers) [inline modules/lang/fortran/autoload.el:127]: An extra branch was necessary here; `abbreviate-file-name` gives an unhelpful error when `project-root` is `nil`.

**Review by hlissner — APPROVED** [review]

**Comment by hlissner** [comment]
LGTM. Thanks again for your help!

**Pull request merged**